src/psygnal/_evented_model_v1.py

```diff
@@ -343,15 +351,44 @@ def _super_setattr_(self, name: str, value: Any) -> None:
         else:
             super().__setattr__(name, value)
 
-    def __setattr__(self, name: str, value: Any) -> None:
-        if (
-            name == "_events"
-            or not hasattr(self, "_events")  # can happen on init
-            or name not in self._events
-        ):
-            # fallback to default behavior
-            return self._super_setattr_(name, value)
+    def _check_if_values_changed_and_emit_if_needed(self) -> None:
+        """
+        Check if field values changed and emit events if needed.
 
+        The advantage of moving this to the end of all the modifications is
+        that comparisons will be performed only once for every potential change.
+        """
+        if self._delay_check_semaphore > 0 or len(self._changes_queue) == 0:
```

Comment:
Why aren't these covered?

Reply:
I see 100% coverage.

Reply:
yeah, that can happen when the tests aren't complete yet.  these are all covered. resolving